### PR TITLE
fix(VTextField): inc/dec type number in list with arrow keys

### LINKED ISSUE #17083: [Bug Report][3.1.13] VTextField type number inc/dec inside list with arrow keys
### Environment
**Vuetify Version:** 3.1.13
**Vue Version:** 3.2.47
**Browsers:** Edge 111.0.1661.62
**OS:** Windows 10

### Steps to reproduce
Focus text field outside and inside of list and use up/down arrows to increment/decrement value.

### Expected Behavior
Increment/decrement VTextField value (number) with up/down keys.

### Actual Behavior
Arrow keys will navigate through the list

### Reproduction Link
[https://play.vuetifyjs.com/#...](https://play.vuetifyjs.com/#eNrVUs1u2zAMfhVCl7SAba3rgAFBOnTvsFvdg2PRiBbrBxLtLAjy7qMsJ/PSS4Ge6pM/Svx+KL6cxE/vq3FAsRYbQuP7hvBHbQE2Y9l4P/1OwDTazojx5SB9v/APQaexV+AGilohuA56HQnu6OgR7GC2GO4LGJt+QGgbC1sEbVupsA1o0BIqOGjaQROCO8Aej/EqJf9psQ1isTKLsSensK8y/VMtJvpaQBJlmOuM5ZIg+VqYnyul5vBAmvrUuUik7ZSkWDgD5eyK4ODCHnYYWPHKV1NNHzaZMi9cfU6zl+plgRheNyiBvFsbudg5hrEN2hNEpCEda+NdIDhBwA7O0AVnYMXLukqXW2d5w/JKPaUbd1/uE2Pm4G5RiEzAyr76HZ3lJT8lA/V8EGuxhqmSakyccC12RD6upWyV5TYeiB5DZZGk9UY+8zUZBkvaYKmceX6svlbfvkvFYZf1CqMpt/wMEQOz1KJY6JDuju/Qmm+yxEP18Jgl5lrZN9uYNN5wSxYcMZQBreIH5+d6X6abtv9y3Zy9yZbkz7U988h7bffxZtptnCb9cnH5sdATW6J6TYrnQmSnrJ1bxOtf0waY1A==)

### Other comments
Is there any way to disable keyboard navigation in the list (if the text field is focused)?

### PR DESCRIPTION
**Pull request by gitemad**
## Description

VTextField type number is not incremented or decremented with arrow keys while in list.

To solve this, I added onKeydown function which stopImmediatePropagation when the type is number and the pressed key is ArrowUp or ArrowDown.

fixes #17083
## Markup:

```vue
<template>
  <v-app>
    <v-main>
      <p>
        Text field outside of list (type number), value can be inc/decremented with arrow keys
      </p>
      <v-text-field v-model.number="value" type="number" />
      <v-list>
        <v-list-item title="Text field in list, arrow keys don't work here">
      		<v-text-field v-model.number="value" type="number" />
        </v-list-item>
        <v-list-item title="Text field in list, arrow keys don't work here">
      		<v-text-field v-model.number="value" type="number" />
        </v-list-item>
      </v-list>
      <v-list>
        <v-list-item title="Text field in list, arrow keys work here">
      		<v-text-field v-model.number="value" />
        </v-list-item>
        <v-list-item title="Text field in list, arrow keys work here">
      		<v-text-field v-model.number="value" />
        </v-list-item>
      </v-list>
    </v-main>
  </v-app>
</template>

<script setup>
import { ref } from 'vue'

const value = ref(0)
</script>

```

### CODE REVIEW

File: packages/vuetify/src/components/VTextField/VTextField.tsx [hunk 1]
```diff
@@ -151,6 +151,12 @@ export const VTextField = genericComponent<Omit<VInputSlots & VFieldSlots, 'defa
         })
       }
     }
+    function onKeydown (e: KeyboardEvent) {
+      if (props.type.toLowerCase() === 'number' &&
```
Comment (yuwu9145) [inline packages/vuetify/src/components/VTextField/VTextField.tsx:155]: Probably all input keyDown events should stop bubbling up. 

`e.stopPropagation()` should be enough

**Comment by yuwu9145** [comment]
Can you also add a cy test?

**Comment by gitemad** [comment]
> Can you also add a cy test?

Sure.

**Comment by yuwu9145** [comment]
Thanks, if having `if (props.type.toLowerCase() === 'number')`, for 'text' type, it will lose this effect shown in the video - press up/down key will switch between start and end. This is the behaviour in [v2 ](https://codepen.io/yuchaosydney/pen/poxowOr?editors=101)

I still think any keydown event happened on input should be stopped bubbling up

vuetify-textfield-up-down-key

**Comment by gitemad** [comment]
> Thanks, if having `if (props.type.toLowerCase() === 'number')`, for 'text' type, it will lose this effect shown in the video - press up/down key will switch between start and end. This is the behaviour in [v2 ](https://codepen.io/yuchaosydney/pen/poxowOr?editors=101)
> 
> I still think any keydown event happened on input (regardless of type) should be stopped bubbling up
> 
> vuetify-textfield-up-down-key [ vuetify-textfield-up-down-key ](https://user-images.githubusercontent.com/5698884/230372174-0e74583d-83b3-42a2-bf29-7f0c7ca03514.gif) [ ](https://user-images.githubusercontent.com/5698884/230372174-0e74583d-83b3-42a2-bf29-7f0c7ca03514.gif)

Sorry, I misunderstood.
I removed the condition and now any input regardless of type stopped bubbling up.

**Review by yuwu9145 — APPROVED** [review]

**Comment by johnleider** [comment]
I don't think that this is the correct fix. Looking through the code, this is because `v-list` is focusing the input when [using arrow down](https://github.com/vuetifyjs/vuetify/pull/15998/files#diff-ad5335c9b2770535bda4d1447443d3da1b9ba39b8ad1b0d4e06ae942c4ec6814R190).

**Comment by KaelWD** [comment]
I'm thinking that if someone wants to override this behaviour they should just add `@keydown.stop` to the fields themselves.

**Comment by johnleider** [comment]
This Pull Request is being closed due to inactivity.

If you have any additional questions, please reach out to us in our [Discord community](https://community.vuetifyjs.com).

**Pull request closed without merging**